fastjet/internal/delaunay/delaunay_test.go

```diff
@@ -0,0 +1,461 @@
+				// remove points that have been matched from slice
+				// in case there are duplicate points. So that it
+				// wouldn't pass the test when it shouldn't
+				expv = append(expv[:j], expv[j+1:]...)
+				break
+			}
+		}
+		if !ok {
+			t.Fatalf("Point in the Voronoi Diagram P%s not as expected", points[i])
+		}
+	}
+
+	if math.Abs(area-exparea) > tol {
+		t.Fatalf("got=%f voronoi area, want=%f", got, want)
+	}
+	nn, dist := p11.NearestNeighbor()
+	if !nn.Equals(p15) {
+		t.Fatalf("got=N%s nearest neighbor, want=N%s", nn, p15)
+	}
+	expdist := 1.463
+	if math.Abs(dist-expdist) > tol {
+		t.Fatalf("got=%d distance, want=%d", dist, expdist)
+	}
+}
+
+func TestWalkMedium(t *testing.T) {
+	// NewPoint(x, y, id)
+	p1 := NewPoint(-1.5, 3.2, 1)
+	p2 := NewPoint(1.8, 3.3, 2)
+	p3 := NewPoint(-3.7, 1.5, 3)
+	p4 := NewPoint(-1.5, 1.3, 4)
+	p5 := NewPoint(0.8, 1.2, 5)
+	p6 := NewPoint(3.3, 1.5, 6)
+	p7 := NewPoint(-4, -1, 7)
+	p8 := NewPoint(-2.3, -0.7, 8)
+	p9 := NewPoint(0, -0.5, 9)
+	p10 := NewPoint(2, -1.5, 10)
+	p11 := NewPoint(3.7, -0.8, 11)
+	p12 := NewPoint(-3.5, -2.9, 12)
+	p13 := NewPoint(-0.9, -3.9, 13)
+	p14 := NewPoint(2, -3.5, 14)
+	p15 := NewPoint(3.5, -2.25, 15)
+	pE1 := NewPoint(0, 0, -1)
+	pE2 := NewPoint(-2.3, -0.6, -2)
+	pE3 := NewPoint(2, 1.2, -3)
+	pE4 := NewPoint(-2.8, -0.5, -4)
+	ps := []*Point{p1, p2, p3, p4, p5, p6, pE3, pE4,
+		p9, p10, p11, p12, p13, p14}
+	d := WalkDelaunay(ps)
+	d.RemovePoint(pE4)
+	d.InsertPoint(pE1)
+	d.RemovePoint(pE3)
+	d.InsertPoint(p15)
+	d.InsertPoint(pE2)
+	d.RemovePoint(pE1)
+	d.InsertPoint(p7)
+	d.InsertPoint(p8)
+	d.RemovePoint(pE2)
+	ts := d.Triangles()
+	exp := []*Triangle{
+		NewTriangle(p1, p3, p4),
+		NewTriangle(p1, p4, p5),
+		NewTriangle(p1, p5, p2),
+		NewTriangle(p2, p5, p6),
+		NewTriangle(p3, p4, p8),
+		NewTriangle(p3, p8, p7),
+		NewTriangle(p4, p8, p9),
+		NewTriangle(p4, p9, p5),
+		NewTriangle(p5, p9, p10),
+		NewTriangle(p5, p10, p6),
+		NewTriangle(p6, p10, p11),
+		NewTriangle(p7, p8, p12),
+		NewTriangle(p8, p12, p13),
+		NewTriangle(p8, p13, p9),
+		NewTriangle(p9, p13, p10),
+		NewTriangle(p10, p13, p14),
+		NewTriangle(p10, p14, p15),
+		NewTriangle(p10, p15, p11),
+	}
+	got, want := len(ts), len(exp)
+	if got != want {
+		t.Fatalf("got=%d delaunay triangles, want=%d", got, want)
+	}
+	for i := range ts {
+		ok := false
+		for j := range exp {
+			if ts[i].Equals(exp[j]) {
+				ok = true
+				// remove triangles that have been matched from slice
+				// in case there are duplicate triangles. So that it
+				// wouldn't pass the test when it shouldn't
+				exp = append(exp[:j], exp[j+1:]...)
+				break
+			}
+		}
+		if !ok {
+			t.Fatalf("Triangle T%s not as expected", ts[i])
+		}
+	}
+	nn, dist := p11.NearestNeighbor()
+	if !nn.Equals(p15) {
+		t.Fatalf("got=N%s nearest neighbor, want=N%s", nn, p15)
+	}
+	expdist := 1.463
+	if math.Abs(dist-expdist) > tol {
+		t.Fatalf("got=%d distance, want=%d", dist, expdist)
+	}
+}
+
+func BenchmarkNewDelaunay50(b *testing.B) {
```

Comment:
If we are happy to require go1.7, then these can be in a for loop calling subtests rather than separate functions.